add blog post by Nodar (#89)
getsynth/synth: docs/blog/2021-08-04-test-data.md

---
title: Why not to use prod data for testing - and what to do instead
author: Nodar Daneliya
author_title: Founder
author_url: https://github.com/getsynth
author_image_url: https://avatars.githubusercontent.com/u/12720758?v=4
tags: [test data, data generation, best practice, gdpr, pseudonymisation]
description: Some thoughts on why using prod data for testing is still commonplace and why it is time to change this.
image: https://i.imgur.com/mErPwqL.png
hide_table_of_contents: false
---

> I get it. Almost everyone has done it. Your users’ data is sitting there ready to use, just copy-paste that to your dev environment and there you have some good test data for yourself. Job done - easy - you can move on to the next task.

Reviewer: *it* to your dev environment.

This is rarely a valid reason to do anything, yet sometimes we inherit a process that deep down we know is flawed, but perhaps feel like we don't have enough knowledge or authority to challenge.

I've been there. When you are new at a company, telling more senior people that the current way of doing things is flawed seems intimidating, but if you have a viable alternative (more on that below) you can suggest (and ideally deliver on), your co-workers will appreciate it.

### 2. No one will find out

Having been in the data privacy space for a while now, it is painfully obvious how often “they” actually find out. One of the big improvements that GDPR has brought about is that it put a spotlight on how frequent and widespread data leaks and breaches actually are - not a day goes by without new ones being reported; so hoping that this doesn't affect you is not really an ideal strategy in light of this.

My colleague [Andre](https://twitter.com/llogiq), recently posted a [great thread](https://twitter.com/llogiq/status/1395109705148780545) about this.

### 3. It’s just for myself and a bunch of my co-workers, what’s the worst that can happen

> By taking data to less secure environment you waste the efforts that you put into security controls of your production environments. Proper privacy measures go one step beyond and look at what happens if there is a data breach - either due to a mistake, or an attack - by minimising the number of people who work with real (sensitive) data you complement your other data security measures.

Author: ```suggestion
By taking data to a less secure environment you waste all effort you put into security controls of your production environments. Proper privacy measures go one step beyond and look at what happens if there is a data breach - either due to a mistake, or an attack - by minimising the number of people who work with real (sensitive) data you complement your other data security measures.
```

Sensitive data should only really be accessed when absolutely necessary - that’s in the spirit of building proper privacy-respecting culture.

### 4. There is just no f…ing time to do this properly

> It can be really painful to pseudonymise data, especially if your data model is very large and complex. It can take even more time to set up data obfuscation pipelines that keep your test data up to date - as if this wasn’t enough - all these scripts keep breaking as your data model evolves. Basically, in so many cases, the task of obfuscating data while preserving its complexity and referential integrity is just too much hassle. Shameless plug: I have some good news and some better news.

Reviewer: What's the 'better news' here? I take it the good news is Synth.

## Final thoughts

Last but not least, trust is 🔑 - the way you treat your users’ data in a way defines your relationship with them. Being responsible about the data that your users trust you with is incredibly important in building a good and healthy culture for your company or project. Trust is hard to build and easily lost. This may seem trivial, but these “small” decisions define the future of your company, not the “culture” page on your website.

Being responsible is a team effort, but it pays off long-term. [^3]

### Notes

[^1]: [Synth](https://getsynth.com/) goes one step beyond pseudonimisation - it creates completely fake data that looks and quacks like the real data. Data generated with Synth can be shared more freely.

> [^2]: [Article 25](https://gdpr-info.eu/art-25-gdpr/) and [Article 32](https://gdpr-info.eu/art-32-gdpr/) of GDPR refer to the requirements around implementing pseudonymisation.

Author: ```suggestion
[^2]: GDPR [Article 25](https://gdpr-info.eu/art-25-gdpr/) and [32](https://gdpr-info.eu/art-32-gdpr/) refer to the requirements around implementing pseudonymisation.
```

> [^3]: We often get asked what is the right stage of the project/company to start using something like Synth - the answer is the earlier the better. With our declarative “data as code” framework, your test data generation will be evolve alongside your data model.

Author: ```suggestion
[^3]: We often get asked what is the right stage of the project/company to start using something like Synth. The best answer is right from the start, the second best is right now. With our declarative “data as code” framework, your test data generation will evolve alongside your data model.
```
